Orange/canvas/scheme/readwrite.py

```diff
@@ -905,6 +906,165 @@ def indent_(element, level, last):
 
     return indent_(element, level, True)
 
+# Parent nodes are nodes with no inputs
+def get_parent_nodes(links, viable_nodes, adjs):
+    parents = []
+    for node in viable_nodes:
+        parent = True
+        for adj in adjs:
+            if adj[1] == node:
+                parent = False
+                break
+        if parent:
+            parents.append(node)
+    return parents
+
+def get_links_from(parent_node, adjs):
+    matches = []
+    for adj in adjs:
+        if adj[0] == parent_node:
+            matches.append(adj)
+    return matches
+
+def get_links_to(child_node, adjs):
+    matches = []
+    for adj in adjs:
+        if adj[1] == child_node:
+            matches.append(adj)
+    return matches
+
+def scheme_toposort(scheme):
+    """
+    Generates a topographically sorted of nodes to be processed.  This
+    provides a linear method by which the entire scheme can be evaluated and
+    exported to a static Python script.  Disconnected nodes with no
+    connections are ignored.
+
+    """
+    adjs = []
+    for link in scheme.links:
+        adjs.append((link.source_node, link.sink_node,))
```

Comment:
Thank you for this, @PrimozGodec!  Python isn't my primary language, so I'm eager to improve my code with more unique features like this